serialize.go

```diff
@@ -0,0 +1,153 @@
+package sqlite
+
+// #include <sqlite3.h>
+// #include <stdlib.h>
+import "C"
+import (
+	"runtime"
+	"unsafe"
+)
+
+// Serialized contains schema and serialized data for a database.
+type Serialized struct {
+	schema         string
+	data           []byte
+	sqliteOwnsData bool
+	shouldFreeData bool
+}
+
+// NewSerialized creates a new serialized DB from the given schema and data.
+//
+// If copyToSqlite is true, the data will be copied. This should be set to true
+// if this will be used with SQLITE_DESERIALIZE_FREEONCLOSE.
+func NewSerialized(schema string, data []byte, copyToSqlite bool) *Serialized {
+	s := &Serialized{
+		schema: schema,
+		data:   data,
+	}
+	if copyToSqlite {
+		sqliteData := (*[1 << 28]uint8)(unsafe.Pointer(C.sqlite3_malloc(C.int(len(data)))))[:len(data):len(data)]
+		copy(sqliteData, data)
+		s.data = sqliteData
+		s.shouldFreeData = true
+		s.sqliteOwnsData = true
+		runtime.SetFinalizer(s, func(s *Serialized) { s.free() })
+	}
+	return s
+}
+
+// Schema returns the schema for this serialized DB.
+func (s *Serialized) Schema() string {
+	if s.schema == "" {
+		return "main"
+	}
+	return s.schema
+}
+
+// Bytes returns the serialized bytes. Do not mutate this value. This is only
+// valid for the life of its receiver and should be copied for any other
+// longer-term use.
+func (s *Serialized) Bytes() []byte { return s.data }
```

Comment:
I would prefer more robust API semantics for this. This is too error prone. Assume the user of this API is _not_ going to read your docs. How can we make it fool proof. Consider that there is virtually no case where someone is going to instantiate a `Serialized` object and _not_ copy the data. So why bother with the intermediary? Why even have a Serialized object at all? 

IMO the ideal API should just accept a Writer where the serialized data is written, or  accept a Reader where the serialized data can be read from so it can be instantiated into the database.